### DISCUSSION TITLE
[Question][Hw1] Assigning boolean values to variables

### DISCUSSION THREAD

**Discussion by LMMilliken** [prosyslab-classroom/cs424-program-reasoning#289] [Q&A]
Name: Louis Milliken

I am trying to test my `int_of_bool` method and the bool case for `print`, but I don't seem to be able to write C code that would result in a boolean being used or defined without resulting in operators that we don't support being included in the LLVM it (such as zext).

For example, the following C code:
```
int main()
{
  int x = 10;
  int y = input();
  int z = (x == y);
  print(z);
  return 0;
}
```
Would result in the following LLVM IR, which our interpreters would not support:
```
...
define dso_local i32 @main() #0 {
entry:
  %call = call i32 (...) @input()
  %cmp = icmp eq i32 10, %call
  %conv = zext i1 %cmp to i32
  call void @print(i32 %conv)
  ret i32 0
}
...
```
Any advice on how to write C code to test these cases would be appreciated.

**Comment by goodtaeeun** [comment]
You do not have to achieve 100% code coverage for this assignment.
When testing the code, you may encounter some code regions that cannot be tested. For example, there might be a dead code. That is why it is impractical to aim for 100% code coverage in practice.

**Comment by goodtaeeun** [comment]
To give a more direct answer to your question, writing a C code that tests the `int_of_bool` and the bool case of `print` is not possible, at least in the scope of this assignment. Thus, you do not have to worry about it.

**Comment by LMMilliken** [comment]
Thank you for your answer! If C code cannot be written to test these methods, can we then assume that these methods won't be tested in gradescope? Apologies if this is prying too much about the hidden tests

**Comment by goodtaeeun** [comment]
Yes, you can assume that there will be no tests that target `int_of_bool` or the bool case of `print` function. However, this does not mean that we do not have unit tests for other functionalities either.
